iml-manager-cli/src/server.rs

```diff
@@ -373,6 +391,133 @@ pub async fn server_cli(command: ServerCommand) -> Result<(), ImlManagerCliError
+                .iter()
+                .map(|host| TestHostConfig {
+                    address: host.address.as_str(),
+                    auth_type: "existing_keys_choice".into(),
+                })
+                .collect();
+
+            let Objects { objects } = post("test_host", Objects { objects })
+                .await?
+                .error_for_status()?
+                .json()
+                .await
+                .map_err(iml_manager_client::ImlManagerClientError::Reqwest)?;
+
+            let cmds = objects
+                .into_iter()
+                .map(|CmdWrapper { command }| command)
+                .collect();
+
+            term.write_line(&format!("{} preflight checks...", style("Running").green()))?;
+
+            let cmds = wait_for_cmds(cmds).await?;
+
+            let jobs: Vec<_> = cmds
+                .iter()
+                .flat_map(|cmd| cmd.jobs.clone())
+                .filter_map(|job| extract_api_id(&job).map(|x| x.to_string()))
+                .map(|x| ["id__in".into(), x])
+                .chain(iter::once(["limit".into(), "0".into()]))
+                .collect();
+
+            let jobs: ApiList<TestHostJob> = get(TestHostJob::endpoint_name(), jobs).await?;
+
+            tracing::debug!("jobs {:?}", jobs);
+
+            let all_passed = jobs
+                .objects
+                .into_iter()
+                .flat_map(|x| x.step_results.into_iter().map(|(_, b)| b))
+                .fold(true, |state, check| {
+                    if !check.valid {
+                        display_error(format!(
+                            "Host {} has failed pre-flight checks.",
+                            check.address,
+                        ));
+
+                        generate_table(
+                            &["Check", "Passed"],
+                            check.status.iter().map(|x| {
+                                let v = if x.value {
+                                    format_success("Pass")
+                                } else {
+                                    format_error("Fail")
+                                };
+
+                                vec![x.name.clone(), v]
+                            }),
+                        );
+
+                        false
+                    } else {
+                        state
+                    }
+                });
+
+            if !all_passed {
+                return Err(ImlManagerCliError::ApiError(
+                    "Preflight checks failed".into(),
+                ));
+            }
+
+            let host_ids = cmds
+                .into_iter()
+                .flat_map(|cmd| cmd.jobs)
+                .filter_map(|job| extract_api_id(&job).map(|x| x.to_string()));
+
+            let objects = host_ids
+                .map(|address| HostRemoverConfig {
+                    address,
+                    verb: "Remove".to_string(),
+                    auth_type: "existing_keys_choice".into(),
```

Comment:
Have you tested whether `auth_type` and `server_profile` are needed?

I haven't checked, but I'd be surprised if they were required to remove a host...